### PR TITLE
added support for ui-router-1.0.0.rc API changes

### PR DESCRIPTION
**Pull request by justinwatkins**
I added support to use the new $transitions service mechanism for angular ui-router 1.0.  A little refactoring for DRY as well.  Please let me know if there is anything required to accept the PR.

### CODE REVIEW

File: dist/angular-jwt.js [hunk 1]
```diff
@@ -83,17 +90,29 @@ angular.module('angular-jwt.authManager', [])
...
+          if (route && route.data && route.data.requiresLogin === true && !isAuthenticated()) {
             invokeRedirector(config.unauthenticatedRedirector);
+            return false;
           }
-        }
       }
 
-      var eventName = ($injector.has('$state')) ? '$stateChangeStart' : '$routeChangeStart';
-      $rootScope.$on(eventName, verifyRoute);
+      if ($injector.has('$transition')) {
```
Comment (kylemantesso) [inline dist/angular-jwt.js]: I think this should be:
if ($injector.has('$transition**s**')) {

File: dist/angular-jwt.js [hunk 2]
```diff
@@ -83,17 +90,29 @@ angular.module('angular-jwt.authManager', [])
...
+        var $transitions = $injector.get('$transitions');
+        $transitions.onStart({}, verifyState);;
```
Comment (kylemantesso) [inline dist/angular-jwt.js]: Remove extra ;

**Comment by kylemantesso** [comment]
Also, I get the following error when it tries to redirect using $location.path:
TransitionRejection(type: 3, message: The transition has been aborted, detail: Hook aborted transition)

I don't think the redirect can use $location.path from a transition hook

**Comment by kylemantesso** [comment]
I put these suggestions into a pull request on @justinwatkins branch

**Comment by kukac7** [comment]
when it will be updated because it does not work with ui-router 1.x?
thanks!

**Comment by chenkie** [comment]
@justinwatkins thanks for your PR :) 

I can merge this in as soon as we resolve the items that @kylemantesso commented on.

Thanks!

**Comment by timelf123** [comment]
Rebased the fine work done by @kylemantesso and @justinwatkins into https://github.com/auth0/angular-jwt/pull/152

**Pull request merged**